### ISSUE TITLE
pasting code from helpfile uses spaces and not tabs

### ISSUE THREAD

**Issue by telephon** [supercollider/supercollider#1536]
pasting code from helpfile uses spaces and not tabs, which would be the standard setting in the editor. It should use the setting from the language config, I suppose.

**Comment by scztt** [comment]
@telephon - Okay if I label this enhancement / 3.8? If the helpfile uses spaces, pasting will result in spaces, in the same way it would copy/pasting anything else w/ spaces would. I think a feature that automatically converts tabs<->spaces on paste depending on your settings would be *awesome* (this is a constant annoyance for me in almost every IDE I use) - but that definitely requires some design, and probably a preference to be added as minimum, so I think it's out of scope as a 3.7 bugfix?

**Comment by telephon** [comment]
Yes sure – I deliberately didn't add the 3.7 milestone, there's enough to do!

**Comment by telephon** [comment]
The problem is that *even if the helpfile is written with tabs* it secretly converts them to spaces before you paste it.

**Comment by telephon** [comment]
related: #2094.

**Comment by timblechmann** [comment]
the ide doesn't perform any explicit conversion. maybe fire up a debugger and see where the tabs are coming from.

**Comment by telephon** [comment]
Maybe somewhere in the lexer? https://github.com/supercollider/supercollider/blob/master/SCDoc/lex.scdoc.cpp

**Comment by telephon** [comment]
At least it happens somewhere in the conversion of the `.schelp`file to the `.html` file.

E.g. compare

https://github.com/supercollider/supercollider/blob/master/HelpSource/Classes/Condition.schelp
http://doc.sccode.org/Classes/Condition.html

**Comment by telephon** [comment]
@snappizz while you are at it, can you see where this conversion from tabs to spaces happens?

**Comment by telephon** [comment]
I can still reproduce this.

**Comment by telephon** [comment]
Still the case.

**Comment by capital-G** [comment]
This seems to be something how SCDoc handles strings/tabs?

E.g.

https://github.com/supercollider/supercollider/blob/develop/HelpSource/Classes/Float.schelp#L169-L175

uses tabs

<img width="326" alt="Image" src="https://github.com/user-attachments/assets/f878cf6e-592f-43a8-a331-6c2c73e3d976" />

but the HTML `view-source:https://dev.docs.supercollider.online/Classes/Float.html` converts this to spaces already

<img width="310" alt="Image" src="https://github.com/user-attachments/assets/22dadab6-1792-4781-b037-75c5491a9e90" />

I haven't found why sclang would do this when writing the HTML file, so it seems someone would need to dig into the bison parser :/ I pass. This is hopefully something which can be fixed in a rewrite of scdoc.

PS: This is one of the reasons why most programming languages prefer whitespaces to get rid of such ambiguities/issues^^ but not lets get into this discussion...
